internal/characters/alhaitham/skill.go

```diff
@@ -0,0 +1,293 @@
+package alhaitham
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/enemy"
+)
+
+var skillTapFrames []int
+var skillHoldFrames []int
+
+const (
+	skillTapHitmark  = 19
+	skillHoldHitmark = 28
+	projectionICDKey = "alhaitham-projection-icd"
+	particleICDKey   = "alhaitham-particle-icd"
+)
+
+var mirror1HitmarkLeft = []int{39}
+var mirror1HitmarkRight = []int{40}
+
+var mirror2HitmarksLeft = []int{28, 37}
+var mirror2HitmarksRight = []int{26, 35}
+
+var mirror3Hitmarks = []int{32, 41, 51}
+
+var snapshotTimings = []int{20, 22, 26}
+
+func init() {
+	// skill (tap) -> x
+	skillTapFrames = frames.InitAbilSlice(44)
+	skillTapFrames[action.ActionAttack] = 27
+	skillTapFrames[action.ActionSkill] = 28
+	skillTapFrames[action.ActionBurst] = 28
+	skillTapFrames[action.ActionDash] = 33
+	skillTapFrames[action.ActionJump] = 33
+	skillTapFrames[action.ActionSwap] = 36
+
+	// skill (hold)-> x
+	skillHoldFrames = frames.InitAbilSlice(87)
+	skillHoldFrames[action.ActionAttack] = 86
+	skillHoldFrames[action.ActionLowPlunge] = 35
+	skillHoldFrames[action.ActionSkill] = 80
+	skillHoldFrames[action.ActionSwap] = 85
+
+}
+
+func (c *char) Skill(p map[string]int) action.ActionInfo {
+	hold := p["hold"]
+	if hold == 1 {
+		return c.SkillHold()
+	}
+
+	c.Core.Tasks.Add(func() {
+		if c.mirrorCount == 0 { //extra mirror if 0 when cast
+			c.mirrorGain(2)
+			return
+		}
+		c.mirrorGain(1)
```

Comment:
Please consider moving this part out to its own little function to remove some duplication between `Skill` and `SkillHold`.